samples/sample_decode/src/sample_decode.cpp

```diff
@@ -419,6 +434,26 @@ mfxStatus ParseInputString(msdk_char* strInput[], mfxU8 nArgNum, sInputParams* p
         {
             pParams->bSoftRobustFlag = true;
         }
+#if (defined(LINUX32) || defined(LINUX64))
+        else if (0 == msdk_strcmp(strInput[i], MSDK_STRING("-device")))
+        {
+            if (strcmp(pParams->strDevicePath, "") != 0){
+                msdk_printf(MSDK_STRING("error: you can specify only one device\n"));
+                return MFX_ERR_UNSUPPORTED;
+            }
+            if(i + 1 >= nArgNum)
+            {
+                PrintHelp(strInput[0], MSDK_STRING("Not enough parameters for -device key"));
+                return MFX_ERR_UNSUPPORTED;
+            }
+            SIZE_CHECK(strInput[0], (msdk_strlen(strInput[i])+1) > MSDK_ARRAY_LEN(pParams->strDevicePath));
```

Comment:
I think you can get rid of this if switch to ```std::string```

Reply:
We might need a bigger follow up for that since there should be other parameters following the same scheme. Input/output file names... Can we open an issue for this? I think that using std::string here right away is ok (with the later change covering other cases).